File: mira-api/src/ai/providers/outfit-analysis.provider.ts
import { MiraOccasion } from '../contracts/mira-occasion';
import { OutfitAnalysisResult } from '../contracts/outfit-analysis-result.interface';

export const OUTFIT_ANALYSIS_PROVIDER = Symbol('OUTFIT_ANALYSIS_PROVIDER');

export interface OutfitAnalysisProvider {
  analyze(imageBytes: Buffer, occasion: MiraOccasion): Promise<OutfitAnalysisResult>;
}

File: lib/core/ai/providers/outfit_analysis_provider.dart
import '../models/mira_occasion.dart';
import '../models/outfit_analysis_result.dart';

/// Abstraction for outfit AI providers (FASHN.ai, Style DNA, Style.me).
abstract class OutfitAnalysisProvider {
  Future<OutfitAnalysisResult> analyze({
    required List<int> imageBytes,
    required MiraOccasion occasion,
  });
}
